codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/AWSServiceConfig.kt

```diff
@@ -138,20 +139,142 @@ class AWSServiceConfig(writer: SwiftWriter, val ctx: ProtocolGenerator.Generatio
         }
 
         writer.dedent().write("}")
+    }
+
+    fun renderAsyncInitializer(
+        awsConfigs: List<ConfigField>
+    ) {
+        writer.writeDocs("Creates a configuration asynchronously")
+        writer.openBlock("public convenience init(", ") async throws {") {
+            awsConfigs.forEachIndexed { index, config ->
+                val terminator = if (index != awsConfigs.lastIndex) ", " else ""
+                writer.write("${config.memberName}: ${config.paramFormatter}$terminator", config.type)
+            }
+        }
+        writer.indent()
+
+        // Create our file based config
+        writer.write("let \$L = try await CRTFileBasedConfiguration.makeAsync()", FILE_BASED_CONFIG_LOCAL_NAME)
         writer.write("")
 
-        // partitionID computed var
-        writer.openBlock("public var partitionID: String? {", "}") {
-            writer.write("return \"\$L - \\(region ?? \"\")\"", serviceName)
+        // Resolve the region resolver
+        writer.write("let resolvedRegionResolver = try \$L ?? DefaultRegionResolver { _, _ in \$L }", REGION_RESOLVER, FILE_BASED_CONFIG_LOCAL_NAME)
```

Comment:
So instead of calling the async region resolver here during initialization, we pass in the block that resolves region asynchronously & call it at some later time when async is not inconvenient?

Reply:
No, we still resolve the region (which is an async call) in this init. This happens just below this on line 170. 

Here we are creating the region resolver, which is either the region resolver provided by the customer or if that is `nil` then its the `DefaultRegionResolver`. The `DefaultRegionResolver` takes a block as an input. This block returns the file based config that should be used by the region resolver.